doc/developers.rst

```diff
@@ -66,37 +68,82 @@ Install Autosklearn
       
       .. code-block:: console
       
-         $ brew install swig # (or apt-get)
+         (venv) $ brew install swig # (or apt-get)
    
    2. Install gcc (MacOS only)
-   Use your package manager to install gcc (necessary for xgboost)
-   
-      .. code-block:: console
       
-         $ brew install gcc # (or apt-get)
-   
-   3. Install xgboost and autosklearn
+      Use your package manager to install gcc (necessary for xgboost)
    
       .. code-block:: console
       
-         $ export CC=/path/to/installed/gcc; export CXX=/path/to/installed/g++; pip install xgboost
-         $ pip install autosklearn
+         (venv) $ brew install gcc # (or apt-get)
+
+Install all the packages and commit hooks
+   When the project is installed through poetry both project requirements and development requirements are installed. Install commit-hooks using the `pre-commit`_ utility.
+
+   .. _pre-commit: https://pre-commit.com/
+
+   .. code-block:: console
+   
+      (venv) $ poetry install -v
+      (venv) $ poetry run pre-commit install
 
 Making sure everything works
    1. Run pytest to make sure you're good to go
    
       .. code-block:: console
       
-         $ pytest
+         (venv) $ poetry run pytest
    
    2. Run tox to run in supported python versions (optional)
    
       .. code-block:: console
       
-         $ tox # supply the -r flag if you changed the dependencies
+         (venv)$ poetry run tox -r # supply the -r flag if you changed the dependencies
+
+   3. Run make html in foreshadow/doc to build the documentation (optional)
+   
+      .. code-block:: console
+      
+         (venv) $ poetry run make html
    
    If all the tests pass you're all set up!
 
+Suggested development work flow
+   1. Create a branch off of development to contain your change
+   
+      .. code-block:: console
+      
+         (venv) $ git checkout development
+         (venv) $ git checkout -b {your_feature}
+
+   2. Run pytest and pre-commit while developing
+      This will help ensure you haven't horrifically broken something while adding your feature and will help you catch bugs as you develop. Pre-commit will help make sure that your formatting is pristine before create a pull request.
```

Comment:
Are we collectively agreeing to informal language in our documents?

Reply:
I think informal language makes reading the docs easier but I'm open to removing it.